themes/bootstrap3/templates/cart/contents.phtml

```diff
@@ -6,7 +6,7 @@
       <div class="checkbox">
         <label>
           <?=$this->record($record)->getCheckbox('cart')?>
-          <a title="<?=$this->transEscAttr('View Record')?>" href="<?=$this->escapeHtmlAttr($this->recordLinker()->getUrl($record))?>" data-lightbox-ignore><?=$this->escapeHtml($record->getBreadcrumb())?></a>
+          <a id="<?=$this->escapeHtmlAttr($this->record($record)->getUniqueIdWithSource())?>" title="<?=$this->transEscAttr('View Record')?>" href="<?=$this->escapeHtmlAttr($this->recordLinker()->getUrl($record))?>" data-lightbox-ignore><?=$this->escapeHtml($record->getBreadcrumb())?></a>
```

Comment:
I don't think that adding an element ID here in the cart/contents.phtml template is going to help to label buttons rendered by record/cart-buttons.phtml. The contents template only renders when a user opens the cart, whereas the buttons render in the main page as part of the record view or search results. Thus, the IDs being referenced won't exist most of the time when they are referenced. Additionally, if we tried to put these IDs on the search result listings or record view, then we would end up with duplicate IDs.

Reply:
I restored previous changes in result-list, cart-buttons and toolbar (e8e6c79e46b318f9e5b219efc63). Although cart/content could stay, because it applies for the cart page and its checkboxes. What do you think?

Reply:
I think the thing to consider when adding the ID to cart/contents is whether that will lead to duplicate IDs on the page; if we're only going to use cart IDs inside the cart, then it's fine. But if we're going to have them in other places, then we'll run into duplication when we add an item to the cart and then open the cart contents in a lightbox, since the same ID will be present in two different parts of the DOM. Maybe we should consider a parameter to prefix the ID so we can ensure ID uniqueness in different contexts -- i.e. one prefix for associating "add to cart" buttons with search results, and a different prefix for associating cart content checkboxes with cart content listings.

Reply:
Thanks for your advice. I noticed that problem. It should be possible to open record view from result list [()](https://katalog.hmt-leipzig.de/Search/Results?lookfor=lightbox&type=AllFields&unsetLineupSearch=true), so I tend to separate between lightbox and non-lightbox without prefix instead of a contextual differentiation.

Reply:
I can't think of a scenario where favorites and holdings would get rendered side-by-side -- it's not possible in "standard" VuFind, though I suppose it could be done as a local customization somewhere. Using a prefix for lightbox content is probably helpful, as long as we never develop a system with multiple or stacked lightboxes. I still slightly prefer prefixing based on context rather than lightbox status because it seems more semantically meaningful, but it's also probably more work to implement.

If you wanted to account for lightbox status in getUniqueHtmlElementId, I think it could probably be done by accessing the view model from inside the helper so you can check the inLightbox property. I think `$this->getView()` might do the job, but I can't remember if that gives you the view object or only the renderer.

Reply:
It is only the renderer, but by using the view model helper within the record view helper we could get this information and the current template as "context". The lightbox prefix should be useful for the edge case of rendering a modal of the same page (Cart/Home on new tab + click on items). 

I wonder if we should use either $idPrefix OR a context string based on the template as implemented yet in getUniqueHtmlElementId. It seems redundant otherwise.

Reply:
Yes, very good point about the edge case of opening a lightbox that duplicates the current page content!

Creating a prefix based on the template being rendered and the lightbox status might be enough to make everything work correctly, though explicitly passing in a context string instead of relying on the template name might be a little more robust if parts of the code end up getting refactored to smaller sub-templates in future. (Though that comment is not based on a recent examination of how the parts fit together, so maybe I'm not fully understanding what you propose).

Reply:
Using the template name would be more generic, I thought. How should we name the context string from and where should it be set (in the template)? The context variable could be an array as I understand it, that could cause trouble.

Reply:
I was just thinking that if we only have two or three contexts, the template that generates the unique ID would just pass a string like `result_` or `cart_` or `list_` to the call to get the unique HTML ID. If we have distinct templates for all these scenarios, then it's easy to do. If there are any shared templates, then the context string will have to be passed in to the template from a higher level where the context can be identified.